handler.go

```diff
@@ -2057,3 +2084,41 @@ func (h *Handler) handleGetID(w http.ResponseWriter, r *http.Request) {
 }
 
 type defaultClusterMessageResponse struct{}
+
+type queryValidationSpec struct {
+	required []string
+	args     map[string]bool
+}
+
+func QueryValidationSpecRequired(requiredArgs ...string) *queryValidationSpec {
```

Comment:
What do you think about splitting this into `NewQueryValidationSpec()` and `Required()`?
Then we could define something like:
```
h.validators["GET /some/path"] = NewQueryValidationSpec().Optional("index")
h.validators["GET /other/path"] = NewQueryValidationSpec().Required("frame")
```

Reply:
Actually it was that way before, but I've changed it to make it look a bit better:
```go
QueryValidationSpecRequired("index").Optional("frame")
```
instead of
```go
QueryValidationSpec().Required("index").Optional("frame")
```
I like the current way better, but I can change it to another style.

Reply:
No need to change it. This was just a suggestion.